drivers/serial/uart_nrfx_uarte.c

```diff
@@ -82,7 +83,22 @@ static inline NRF_UARTE_Type *get_uarte_instance(struct device *dev)
 static void uarte_nrfx_isr(void *arg)
 {
 	struct device *dev = arg;
-	const struct uarte_nrfx_data *data = get_dev_data(dev);
+	struct uarte_nrfx_data *data = get_dev_data(dev);
+	NRF_UARTE_Type *uarte = get_uarte_instance(dev);
+
+	if (data->disable_tx_irq &&
+	    nrf_uarte_event_check(uarte, NRF_UARTE_EVENT_ENDTX)) {
+		nrf_uarte_int_disable(uarte, NRF_UARTE_INT_ENDTX_MASK);
+
+		/* If there is nothing to send, driver will save an energy
```

Comment:
How is this code saving more energy than what would anyway be the case if the shift register was empty?